### PR TITLE
Create offers endpoint and offers page

### PR DESCRIPTION
**Pull request by QianyingLuo**

### CODE REVIEW

File: src/repository/crud/product.py [hunk 1]
```diff
@@ -47,3 +47,18 @@ def get_products_by_type(product_type: str):
     except Error as e:
         print(f"Error al obtener productos del tipo {product_type}:", e)
         return []
+
+def get_discounted_products():
+    try:
+        discounted_products: list[product_model.Product] = []
+        cursor = mysql_connection.cursor(dictionary=True)
+        cursor.execute("SELECT * FROM product WHERE discounted_price != 0")
```
Comment (devs-toni) [inline src/repository/crud/product.py]: which is the field that you save in BBDD? discounted_price? is better to save a discount as a percentage and you can calculate it in the function

File: src/repository/crud/user.py [hunk 2]
```diff
@@ -9,11 +9,8 @@ async def save(user: User):
         query = "INSERT INTO user (username, password, email, full_name, address, phone) VALUES (%s, %s, %s, %s, %s, %s)"
         user_data = (
             user.username,
-            user.password,
             user.email,
```
Comment (devs-toni) [inline src/repository/crud/user.py:11]: Why do you remove this fields? Is only for know it . . .

File: src/repository/crud/product.py (hunk 1 above)
Comment (QianyingLuo) [reply src/repository/crud/product.py]: Just replaced the discounted_price with discount_percentage, thanks

File: src/repository/crud/user.py (hunk 2 above)
Comment (QianyingLuo) [reply src/repository/crud/user.py:11]: Finally I replaced the "username, password, email, full_name, address, phone" in user table with "name, email, password, address, phone", in which the address and phone field are optional

**Pull request merged**